### ISSUE TITLE
New feature: calendar sync

### ISSUE THREAD

**Issue by mfsilva22** [PrairieLearn/PrairieTest-feedback#10]
Sync scheduled tests with Google calendar (request from a student on Reddit)

**Comment by josephshepin** [comment]
Yes,  **everybody** wants this! Having a simple .ICS generated would make it easy for us to add these to our calendar of choice

**Comment by josephshepin** [comment]
I know a lot of people don't use outlook as their primary calendar, and use google calendar. I'm not sure if this would be included in the feature, but it's possible to generate a link which creates a google calendar event when clicked with all the event details prefilled. I just got such a link from Peoria Charter to do this and it was magical. The link goes to google calendar specifically, which i'm vouching for here in addition to outlook. Thanks for working on this!

**Comment by nwalters512** [comment]
I think "add to calendar" links for the main platforms (Outlook + Google) would be a nice stepping stone towards full sync. Even a generic option to download an `.ics` file for a particular reservation would probably be sufficient.

The really nice thing to do would be to serve a `webcal://` URL to an ICS file that can be queried for updates.

This approach as a whole ("add to calendar" buttons, downloadable ICS for a specific event, subscribable ICS for an account as a whole) is what Partiful uses. They're the only thing I have hooked up to sync into my calendar 🙂
